src/app/icd/ICDStateObserver.h

```diff
@@ -33,6 +33,7 @@ class ICDStateObserver
     virtual ~ICDStateObserver() {}
     virtual void OnEnterActiveMode()  = 0;
     virtual void OnTransitionToIdle() = 0;
+    virtual void OnICDModeChange()    = 0;
```

Comment:
I think we should change these function to have default implementations that do nothing. It avoid child classes to have empty implementations for the function.

Reply:
I discussed with Louis-Philip about, that. For now since there is only 3 functions, we prefer to leave it without a default implementation. It enforces/informs users of what should be implemented.